test/test_client_reconnect.cpp

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
I'm commenting this test out for now because it's not properly mocking the localhost listener, and the server needs to be created manually in order to for the test to work